src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -569,7 +575,8 @@ JvmtiVTMSTransitionDisabler::VTMS_vthread_mount(jobject vthread, bool hide) {
     VTMS_mount_begin(vthread);
   } else {
     VTMS_mount_end(vthread);
-    if (JvmtiExport::should_post_vthread_mount()) {
+    if (JvmtiExport::can_support_virtual_threads() &&
+        JvmtiExport::should_post_vthread_mount()) {
```

Comment:
It seems odd that "can_support" can be false when "should_post" is true. I would think that "should_post" would always be false when "can_support" is false, and therefore there would be no need to check "can_support".

Reply:
Right, thanks. It is why this check was missed in the first place. Will undo this change.

Reply:
Fixed.